### DISCUSSION TITLE
Custom User.IsAdmin()?

### DISCUSSION THREAD

**Discussion by NotLazy** [gogs/gogs#6862] [Q&A] [answered]
I'm trying to make a small edit to users, to display a navigation item only when the user has a (new) col in the db set to 1, similar to the is_admin col in the db.

How can I go about this? I tried adding `canAccess bool` to `internal/db/user.go:82` (immediately below `IsAdmin bool`) and adding `c.data["canAccess"] = c.User.canAccess` to `internal/context/context.go:267` (immediately below `c.Data["IsAdmin"] = c.User.IsAdmin`) but when I try to build I get:

```
internal/context/context.go:267:41: c.User.canAccess undefined (type *db.User has no field or method canAccess)
```

What am I missing? I'm not 100% familiar with go, but I'm familiar with a handful of other (slightly similar) languages, so forgive me if this is an obvious thing.

**Comment by unknwon** [comment, marked as answer]
Hey! In Go, you have to use capitalize for public fields, so I think it should be `CanAccess`.

**Reply by NotLazy** [reply]
Thanks, this fixed the error and I was able to build, but then when I change can_access in the DB from 0 to 1 (similarly to how is_admin appears to work), it doesn't appear to reflect.

I'm attempting to use this in the head template where `{{ .IsAdmin }}` allows the "admin panel" to show up in the navigation, but using it to display the navigation item using `{{ .CanAccess }}` but no matter if I set the tinyint in mysql to 0 or 1, the navigation item doesn't show up. How can I debug this, or can you give me any pointers?

**Reply by unknwon** [reply]
How have you debugged to ensure the `CanAccess` field is actually reading data from the MySQL?

**Reply by NotLazy** [reply]
That's kinda what I asked. As I said I'm familiar with other languages, but I'm not very familiar with Go

How can I debug Go to determine if it's reading MySQL?

**Reply by unknwon** [reply]
Looks like you're changing `internal/context/context.go:267`, you can print the value one line below:

```go
fmt.Println("c.User.CanAccess", c.User.CanAccess)
```

Recompile and run, you should see the value being printed in the console.
